### PR TITLE
Creates new write path unit_test_build_path pointing to tests/unit in build directory

### PR DESCRIPTION
**Pull request by isha1810**
### Types of changes:

- [ x] Bugfix
- [ ] New feature
- [ ] Refactor

### Component:

- [x ] Code
- [ ] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests and `clang-tidy`.
  For instructions on how to perform the CI checks locally refer to the [Dev
  guide on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

File: tests/Unit/Framework/SetupLocalPythonEnvironment.cpp
Comment (nilsdeppe) [inline tests/Unit/Framework/SetupLocalPythonEnvironment.cpp]: This is probably my fault: you need to mark it static in the `hpp` only... And should probably do it for both the 3 and 2 versions

**Comment by nilsvu** [comment]
@isha1810 @nilsdeppe would you be able to move this PR forward? I think all it really needs is a rebase. If you @isha1810 don't have time for this, would you mind if @nikwit or I take this over?

**Comment by isha1810** [comment]
@nilsleiffischer @nilsdeppe made changes, let me know if this is ok for me to squash.

**Review by nilsdeppe — COMMENTED** [review]
LGTM, please go ahead and squash once @nilsleiffischer has had a chance to take a look at the fixups. Please rebase on develop when you squash since there have been a lot of changes recently

**Comment by isha1810** [comment]
I rebased before I made the changes (this morning). Have there been more changes today?

**Comment by nilsdeppe** [comment]
ah okay, great. Nope! Just over the last week or two :)

**Review by nilsvu — COMMENTED** [review]
LGTM, please squash

**Comment by isha1810** [comment]
@nilsdeppe @nilsleiffischer finished squashing.

**Comment by nilsvu** [comment]
@isha1810 would you mind rebasing on the latest develop branch again? It turns out #2728 was merged right after you rebased, which added an automated test that is required to pass before PRs get merged (that's why the list of checks below reports the "gcc-10 Release" test as pending).

**Comment by isha1810** [comment]
@nilsdeppe @nilsleiffischer finished rebasing

**Review by nilsdeppe — APPROVED** [review]

**Review by nilsvu — APPROVED** [review]

**Pull request merged**
